### PR TITLE
OHRI-2223: Implement the TB app schema based workflow

### PR DESCRIPTION
**Pull request by lucyjemutai**
## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes the ticket number in the format `OHRI-123 My PR title`.
- [ ] My work includes tests or is validated by existing tests.

## Summary
Implement the TB app schema based workflow

## Screenshots

[Screencast from 10-06-2024 03:07:04 ALASIRI.webm](https://github.com/UCSF-IGHS/openmrs-esm-ohri/assets/130601439/ef3399fa-c1a5-4b45-baaa-dadbe7754c81)

image

image

### CODE REVIEW

File: packages/esm-commons-lib/src/utils/encounter-list-config-builder.ts [hunk 1]
```diff
@@ -75,7 +76,26 @@ export const getTabColumns = (columnsDefinition: Array<ColumnDefinition>) => {
     key: column.id,
     header: column.title,
     concept: column.concept,
-    getValue: (encounter) => {
+    getValue: async (encounter) => {
+      if (column.concept) {
```
Comment (hadijahkyampeire) [inline packages/esm-commons-lib/src/utils/encounter-list-config-builder.ts:80]: We shouldn't do this here, we should handle it how we are handling ART reason and ART date

File: packages/esm-commons-lib/src/utils/encounter-list-config-builder.ts [hunk 2]
```diff
@@ -75,7 +76,26 @@ export const getTabColumns = (columnsDefinition: Array<ColumnDefinition>) => {
...
+        switch (column.concept) {
+          case 'treatmentId':
+            return await getTbTreatmentId(encounter, 'treatmentId');
```
Comment (hadijahkyampeire) [inline packages/esm-commons-lib/src/utils/encounter-list-config-builder.ts:83]: We don't want to have specific names like getTB in commons lib, we should make a generic function that handle these cases.

File: packages/esm-tb-app/src/views/program-management/tb-program-management-config.json [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+          {
+            "id": "tbTreatmentId",
+            "title": "TB Treatment ID",
+            "concept": "treatmentId"
+          },
+          {
+            "id": "treatmentStartDate",
+            "isDate": true,
+            "title": "Treatment Start Date",
+            "concept": "tBTreatmentStartDateConcept"
```
Comment (hadijahkyampeire) [inline packages/esm-tb-app/src/views/program-management/tb-program-management-config.json:30]: Is this working? it is not a uuid, I know we are moving to this implementation next but I am not sure it works now.

**Review by hadijahkyampeire — COMMENTED** [review]
Thanks @lucyjemutai , I have left comments concerning the functions

**Comment by lucyjemutai** [comment]
Thank you @hadijahkyampeire , Kindly review again. Thank you

File: packages/esm-tb-app/src/views/program-management/maternal-health.component.tsx [hunk 4]
```diff
@@ -1,27 +1,37 @@
...
-import MdrTbList from './tabs/mdr-tb.component';
-import TbPatientTracing from './tabs/tb-patient-tracing.component';
-import { PatientChartProps } from '@ohri/openmrs-esm-ohri-commons-lib';
+import { PatientChartProps, getMenuItemTabConfiguration, EncounterList } from '@ohri/openmrs-esm-ohri-commons-lib';
+import tptProgramManagemetConfigSchema from './tb-program-management-config.json';
+interface OverviewListProps {
+  patientUuid: string;
+}
 
 const ProgramManagementSummary: React.FC<PatientChartProps> = ({ patientUuid }) => {
```
Comment (hadijahkyampeire) [inline packages/esm-tb-app/src/views/program-management/maternal-health.component.tsx:10]: Why is this component named maternal-health.component.tsx shouldn't be something like tb-program-management.component.tsx

**Review by hadijahkyampeire — COMMENTED** [review]
Thanks @lucyjemutai , I noticed your have fixed the wrong form that we had on dev, thanks. I have some other comments.
1. We need to also make json files for the `Patient Summary` tables in both TB and TPT. They are using the Encounter list and have that column definition that we want to move to the commons lib.
2. The program management file in TB should be named well.
3. Treatment and Follow up and TB contact list are not tabs, you can just leave them as tables, check what I did for the visit summary under hct. check the comparisons below.
Screenshot 2024-06-12 at 00 37 51
Screenshot 2024-06-12 at 00 36 32
Screenshot 2024-06-12 at 00 36 18

Screenshot 2024-06-12 at 00 38 05

**Comment by lucyjemutai** [comment]
image

File: packages/esm-commons-lib/src/utils/summary-card-config-builder.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        column.isTrueFalseConcept,
+        column.type,
+        column.fallbackConcepts,);
+    },
+    getObsSummary: () => { }
+
+  }));
+
+  return columns;
+};
```
Comment (hadijahkyampeire) [inline packages/esm-commons-lib/src/utils/summary-card-config-builder.ts:24]: Add a new line at the end of the file, eslint will flag once we config it.

File: packages/esm-commons-lib/src/utils/summary-card-config-builder.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+      if (column.isConditionalConcept) { return getConditionalConceptValue(encounter, column.conditionalConceptMappings, column.isDate); }
+      return getObsFromEncounter(
+        encounter,
+        column.concept,
+        column.isDate,
+        column.isTrueFalseConcept,
+        column.type,
+        column.fallbackConcepts,);
+    },
+    getObsSummary: () => { }
```
Comment (hadijahkyampeire) [inline packages/esm-commons-lib/src/utils/summary-card-config-builder.ts:19]: Also this should be implemented, we will need for other cards.

**Review by hadijahkyampeire — COMMENTED** [review]
This looks good just some minor suggestions mostly about leaving an empty line at the end of the file, its a better coding practice.

**Comment by lucyjemutai** [comment]
I have updated for the TPT as well..

**Review by hadijahkyampeire — APPROVED** [review]
LGTM, thanks @lucyjemutai

**Pull request merged**